.github/workflows/kafka_demo_test.yml

```diff
@@ -0,0 +1,121 @@
+name: Kafka Demo Test 
+
+on:
+  pull_request:
+    branches: 
+      - main
+  workflow_call:
+    inputs:
+      sidecar-registry:
+        description: "The registry to get the image from"
+        required: true
+        default: "mcr.microsoft.com"
+        type: string
+      key-release-image:
+        description: "The image of the SKR sidecar to use"
+        required: true
+        default: "aci/skr:2.7"
+        type: string
+      consumer-image: 
+        description: "Consumer Image"
+        required: true
+        type: string
+      producer-image: 
+        description: "Producer Image"
+        required: true
+        type: string
+      skr-kafka-demo-topic:
+        description: "Kafka topic"
+        required: true
+        default: "kafka-demo-topic"
+        type: string  
+  workflow_dispatch:
+    inputs:
+      sidecar-registry:
+        description: "The registry to get the image from"
+        required: true
+        type: choice
+        default: mcr.microsoft.com
+        options: 
+          - mcr.microsoft.com
+          - confidentialsidecars.azurecr.io
+      key-release-image:
+        description: "The image of the SKR sidecar to use"
+        required: true
+        default: "aci/skr:2.7"
+        type: string
+      consumer-image: 
+        description: "Consumer Image"
+        default: acc/samples/kafka/consumer:1.0
+        required: true
+        type: string
+      producer-image: 
+        description: "Producer Image"
+        default: acc/samples/kafka/producer:1.0
+        required: true
+        type: string
+  merge_group:
+    branches:
+      - main
+
+jobs:
+  prepare:
+    name: Extract Environment Variables
+    runs-on: ubuntu-latest
+    outputs:
+      key-release-image: ${{ steps.extract-envs.outputs.key-release-image }}
+      consumer-image: ${{ steps.extract-envs.outputs.consumer-image }}
+      producer-image: ${{ steps.extract-envs.outputs.producer-image }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Extract Envs 
+        id: extract-envs
+        run: | 
+          echo "Extracting Environment Variables"
+          echo "Triggering event is ${{ github.event_name }}"
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            echo "key-release-image=mcr.microsoft.com/aci/skr:2.7" >> $GITHUB_OUTPUT
+            echo "consumer-image=mcr.microsoft.com/acc/samples/kafka/consumer:1.0" >> $GITHUB_OUTPUT
+            echo "producer-image=mcr.microsoft.com/acc/samples/kafka/producer:1.0" >> $GITHUB_OUTPUT
+          else
+            echo "key-release-image=${{ format('{0}/{1}', inputs.sidecar-registry, inputs.key-release-image) }}" >> $GITHUB_OUTPUT
+            echo "consumer-image=${{ format('{0}/{1}', inputs.sidecar-registry, inputs.consumer-image) }}" >> $GITHUB_OUTPUT
```

Comment:
This won't work if the user chooses the confidentialsidecars acr, since the kafka demo images aren't being pushed there. Since these images shouldn't change as much as the sidecars, we should use a separate registry as input for them. Currently they should exist publically in the accsamplesmcr. So I suggest we just hardcode the location for these since they won't change much and keep the sidecars input since we will need to test any updates to skr,